source/common/upstream/load_balancer_impl.cc

```diff
@@ -437,20 +432,32 @@ void RoundRobinLoadBalancer::refresh(uint32_t priority) {
       }
     }
 
+    // Compute schedule offset if we already have been balancing for this
+    // locality.
+    const uint64_t offset =
+        hosts.size() == 0
+            ? 0
+            : (scheduler_.find(source) != scheduler_.end() ? scheduler_[source].rr_index_ : seed_) %
```

Comment:
when not using weights, is there a reason that we don't always use a random offset?

Reply:
Yeah; if you use the last offset we had for a locality, we have the nice property that if we update twice with the same host set, we continue the RR where we left of. This avoids biasing towards the earlier hosts in the schedule.